src/lsstseries/utils/column_mapper/column_mapper.py

```diff
@@ -0,0 +1,200 @@
+            Identifies which column contains the flux/magnitude information
+        err_col: 'str', optional
+            Identifies which column contains the flux/mag error information
+        band_col: 'str', optional
+            Identifies which column contains the band information
+        nobs_col: list of 'str', optional
+            Identifies which columns contain number of observations for each
+            band, if available in the input object file
+        nobs_tot_col: 'str', optional
+            Identifies which column contains the total number of observations,
+            if available in the input object file
+        provenance_col: 'str', optional
+            Identifies which column contains the provenance information, if
+            None the provenance column is generated.
+
+        Returns
+        -------
+        ColumnMapper object
+        """
+
+        self.map = {
+            "id_col": id_col,
+            "time_col": time_col,
+            "flux_col": flux_col,
+            "err_col": err_col,
+            "band_col": band_col,
+            "provenance_col": provenance_col,
+            "nobs_total_col": nobs_total_col,
+            "nobs_band_cols": nobs_band_cols,
+        }
+
+        # Specifies which column mappings must be set for the Ensemble
+        self.required = {
+            "id_col": True,
+            "flux_col": True,
+            "time_col": True,
+            "err_col": True,
+            "band_col": True,
+            "provenance_col": False,
+            "nobs_total_col": False,
+            "nobs_band_cols": False,
+        }
+
+        self.known_maps = {"ZTF": ZTFColumnMapper}
+
+    def _set_known_map(self):
+        """Must be defined in a known map class"""
+        raise NotImplementedError
+
+    @staticmethod
+    def map_id() -> str:
+        return None
+
+    def use_known_map(self, map_id):
+        """Use a known mapping scheme
+
+        Parameters
+        ----------
+        map_id: 'str'
+            Identifies which mapping scheme to use
+
+        Returns
+        -------
+        A ColumnMapper subclass object dependent on the map_id provided, for example
+        ZTFColumnMapper in the case of "ZTF"
+
+        """
+        if map_id in self.known_maps.keys():
+            if map_id == "ZTF":
+                self = ZTFColumnMapper()._set_known_map()
+            return self
+        else:
+            raise ValueError(f'Unknown Mapping: "{map_id}"')
+
+    def is_ready(self, show_needed=False):
+        """shows whether the ColumnMapper has all critical columns assigned
+
+        Parameters
+        ----------
+        show_needed: 'bool', optional
+            Indicates whether to also return a list of missing columns
+
+        Returns
+        -------
+        `bool` or tuple of (bool, list) dependent on show_needed parameter
+
+        """
+
+        # Grab required column keys
+        required_keys = [item[0] for item in self.required.items() if item[1]]
```

Comment:
This is a little difficult to read, and requires that the reader know that `self.required` is a dictionary of `str : bool` key/values. Certainly feels like overkill to introduce a new miniclass `Column` with a `name` and `is_required`. But I do wonder if there's a way to make this more intuitive for future devs.

One option that is much more lightweight than a whole class or data class is a named tuple. I looked over the code and didn't immediately see places where we make use of the properties of a dictionary. Perhaps you could consider in the `__init__` method:

```
from collections import namedtuple
Column = namedtuple('Column', ['name', 'is_required'])

self.required = [
            Column(name="id_col", is_required=True),
            Column("flux_col", True),  # `name` and `is_required` aren't necessary to instantiate
            ...
        ]
```

Then here, you can do this:
```
required_keys = [col.name for col in self.required if col.is_required]
```